E2_ESRGAN/lib/train.py

```diff
@@ -0,0 +1,193 @@
+import time
+import logging
+import itertools
+from functools import partial
+import tensorflow as tf
+from lib import utils, model, dataset
+
+
+class Training(object):
+  @classmethod
+  def setup_training(cls, summary_writer, settings, data_dir=None):
```

Comment:
I think its better to just make this instance-specific. That way, a client could use this trainer to train multiple models in parallel without possibility of side-effects.